File: civitasruby/lib/casilla_juez.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
class CasillaJuez < Casilla
  def initialize(nombre,carcel)
    super(nombre)
    @carcel = carcel
  end
  
  def recibe_jugador(actual,todos)
    if(jugador_correcto(actual,todos))
      super(actual,todos)
      todos[actual].encarcelar(@carcel)
    end
  end
  
  def to_string
    cadena="Casilla: #{@nombre} \n"
    cadena+="\n"
    return cadena
  end
  public :to_string

end
end

File: JavaCivitas/src/civitas/CasillaJuez.java
package civitas;

import java.util.ArrayList;

public class CasillaJuez extends Casilla{

    private static int carcel;

    CasillaJuez(int numCasillaCarcel, String nom){

        super(nom);
        carcel = numCasillaCarcel;

    }

    @Override
    public void recibeJugador(int iactual, ArrayList<Jugador> todos){

        if(jugadorCorrecto(iactual, todos)){
            todos.get(iactual).encarcelar(carcel);
            super.informe(iactual,todos);
            System.out.println("El jugador se va a la carcel");
        }

    }

    @Override
    public String toString(){

        return "\n[Casilla] -> " + super.getNombre() +"\n[Tipo] -> Juez\n[Carcel] -> " + carcel +"\n";

    }

}
